.gitlab-ci-agent.yml

```diff
@@ -655,92 +649,175 @@ pre_release_cluster_agent_image:
+
+# Primary destroy steps for all the molecule builds
+.molecule_destroy: &molecule_destroy
+  stage: cleanup
+  script:
+    - cd $CI_PROJECT_DIR/test
+    - ./molecule3.sh --base-config ./molecule/$MOLECULE_SCENARIO/provisioner.setup.yml destroy --scenario-name $MOLECULE_SCENARIO
 
-molecule_tests_compose:
-  <<: *molecule_base
+# Job Seq: Molecule
+# Type: Compose
+
+.molecule_compose_rules: &molecule_compose_rules
+  rules:
+    - if: ($CI_COMMIT_MESSAGE =~ /\[molecule\-compose]/ || $CI_COMMIT_MESSAGE !~ /\[molecule\-/) && $CI_JOB_NAME =~ /molecule_compose_create/
+      when: always
+    - if: ($CI_COMMIT_MESSAGE =~ /\[molecule\-compose]/ || $CI_COMMIT_MESSAGE !~ /\[molecule\-/) && $CI_JOB_NAME =~ /molecule_compose_test/
+      when: on_success
+    - if: ($CI_COMMIT_MESSAGE =~ /\[molecule\-compose]/ || $CI_COMMIT_MESSAGE !~ /\[molecule\-/) && $CI_JOB_NAME =~ /molecule_compose_failure_cleanup/
+      when: on_failure
+    - if: ($CI_COMMIT_MESSAGE =~ /\[molecule\-compose]/ || $CI_COMMIT_MESSAGE !~ /\[molecule\-/) && $CI_JOB_NAME =~ /molecule_compose_cleanup/
+      when: always
+    - if: $CI_COMMIT_MESSAGE =~ /\[molecule\-/
+      when: never
+
+molecule_compose_create:
+  <<: *molecule
+  <<: *molecule_create
+  <<: *molecule_compose_rules
+  variables:
+    MOLECULE_SCENARIO: compose
+    MOLECULE_EPHEMERAL_DIRECTORY: $CI_PROJECT_DIR/tmp_molecule_inventory
+
+molecule_compose_test:
+  <<: *molecule
+  <<: *molecule_test
+  <<: *molecule_compose_rules
+  variables:
+    MOLECULE_SCENARIO: compose
+    MOLECULE_EPHEMERAL_DIRECTORY: $CI_PROJECT_DIR/tmp_molecule_inventory
+  dependencies:
+    - molecule_compose_create
   needs:
     - pre_release_main_agent_image
-  script:
-    - cd $CI_PROJECT_DIR/test && ./molecule3.sh test -s compose
-  artifacts:
-    when: always
-    paths:
-      - $CI_PROJECT_DIR/test/molecule-role/molecule/compose/topic-*.json
-      - $CI_PROJECT_DIR/test/molecule-role/molecule/compose/*.log
-      - $CI_PROJECT_DIR/tmp_molecule_inventory
-    expire_in: 1 week
+    - molecule_compose_create
 
-molecule_tests_compose_cleanup:
-   <<: *molecule_base
-   stage: acceptance_cleanup
-   script:
-     - cd $CI_PROJECT_DIR/test && ./molecule3.sh destroy -s compose
-   needs:
-     - molecule_tests_compose
-   when: on_failure
-   allow_failure: true
+molecule_compose_failure_cleanup:
+  <<: *molecule_destroy
+  <<: *molecule_compose_rules
+  variables:
+    MOLECULE_SCENARIO: compose
+  needs:
+    - molecule_compose_create
 
-molecule_tests_integrations:
-  <<: *molecule_base
+molecule_compose_cleanup:
+  <<: *molecule_destroy
+  <<: *molecule_compose_rules
+  variables:
+    MOLECULE_SCENARIO: compose
+  dependencies:
+    - molecule_compose_test
+    - molecule_compose_create
+  needs:
+    - molecule_compose_test
+    - molecule_compose_create
+
+# Job Seq: Molecule
+# Type: Integrations
+
+.molecule_integrations_rules: &molecule_integrations_rules
```

Comment:
it looks a bit weird to me that we have 4 conditions defined here, that fire based on where they are included.. why not split out the rules over the jobs?

I cannot decypher the commit message matching logic, what runs under what conditions with the commit message? Best add a comment to explain that (maybe even the README.md if it is a feature)

Reply:
I found the comment on .gitlab-ci.yml, that looks great! Still the logic looks convoluted, why 4 conditionals all together?

Reply:
True true, I might be able to make these rules more compact. Looking at this now